### PR TITLE
Base XPM and image cache, dist into .gitignore

### PR DESCRIPTION
**Pull request by Strae**
**Merge request checklist**

- [x] I read the [guidelines for contributing](https://github.com/mistic100/Photo-Sphere-Viewer/blob/master/.github/CONTRIBUTING.md)
- [x] I created my branch from `dev` and I am issuing the PR to `dev`
- [x] I **didn't** commited files in the `dist` directory

### CODE REVIEW

**Comment by mistic100** [comment]
If you ignore "dist" I won't be able to publish to Bower

---

Can you explain what's going on here ? And how it works. As far as I know, the browser will cache the image and will not download it twice, adding a custom layer of cache is very error prone and useless.

Also you duplicated a whole lot of code which I don't accept.

**Comment by mistic100** [comment]
And and find this if syntax (expected == actual) very hard to read, I prefer the other war arround

File: src/js/PhotoSphereViewer.js [hunk 1]
```diff
@@ -157,7 +157,12 @@ function PhotoSphereViewer(options) {
       cropped_height: 0,
       cropped_x: 0,
       cropped_y: 0
-    }
+    },
+    usexmpdata: false, // most panoramas doesnt have xmpdata, and checking for them require to download the image twice (???)
```
Comment (mistic100) [inline src/js/PhotoSphereViewer.js]: this assertion is false and you don't use this property

File: src/js/PhotoSphereViewer.core.js [hunk 2]
```diff
@@ -163,13 +174,101 @@ PhotoSphereViewer.prototype._loadTexture = function() {
...
+PhotoSphereViewer.prototype._preloadPanorama = function(pano, progressCallback) {
+  var self = this;
+  var defer = D();
+  var loader = new THREE.ImageLoader();
+  var progress = 0;
+
+  loader.setCrossOrigin('anonymous');
+
+  var onload = function(img) {
+    if (self.loader) {
```
Comment (mistic100) [inline src/js/PhotoSphereViewer.core.js]: I guess you don't want to update the visual loader while preloading ?

File: src/js/PhotoSphereViewer.core.js [hunk 3]
```diff
@@ -163,13 +174,101 @@ PhotoSphereViewer.prototype._loadTexture = function() {
...
+      self.loader.setProgress(100);
+    }
+
+    var pano_data = {
```
Comment (mistic100) [inline src/js/PhotoSphereViewer.core.js]: If you don't call `_loadXMP` (why ?) at least use `config.pano_data`

File: src/js/PhotoSphereViewer.core.js (hunk 2 above)
Comment (Strae) [inline src/js/PhotoSphereViewer.core.js]: The idea is to show the marker in the view, but 'disabled'
And as soon as the panorama get loaded, the marker become 'enabled'

File: src/js/PhotoSphereViewer.js (hunk 1 above)
Comment (Strae) [inline src/js/PhotoSphereViewer.js]: I've got panorama photos made with different smartphones, with ricoh theta and other cameras; never found xmp details.. 
Anyway, while using `usexmpdata` the image file get loaded twice

**Comment by Strae** [comment]
Sorry mate, I was in the rush and published the pull request too early, give me the time to implement better the preload feature and i'll resubmit the pull.
Feel free to close this request

**Comment by mistic100** [comment]
And still, nothings prove the file is loaded twice.



there are two requests but the second one is handled by the browser cache.

**Pull request closed without merging**